### ISSUE TITLE
trivially_copy_pass_by_ref not firing with code emitted from proc macro

### ISSUE THREAD

**Issue by winterqt** [rust-lang/rust-clippy#7807]
Lint name: `trivially_copy_pass_by_ref`

I tried this code:

```rust
use proc_macro::TokenStream;

#[proc_macro_derive(Test)]
pub fn test(_: TokenStream) -> TokenStream {
    "fn b(x: &u32) {}".parse().unwrap()
}

#![deny(clippy::trivially_copy_pass_by_ref)]

// fails:
// fn a(x: &u32) {}

#[derive(clippywhy::Test)]
struct A {}

fn main() {}
```

I expected to see this happen: for the lint to fire, as it does for `a`.

Instead, this happened: the lint did not fire.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.55.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: aarch64-apple-darwin
release: 1.55.0
LLVM version: 12.0.1
```

**Comment by winterqt** [comment]
This would be why:

https://github.com/rust-lang/rust-clippy/blob/3d9c4a655b04e9616ded8374dc61cb9108f4ae06/clippy_lints/src/pass_by_ref_or_value.rs#L247

Unsure if this would be classified as a bug or not, as this was added with the initial(?) implementation of this lint back in 50419118b41b0723272517ef9be3baa3fb37dc3c.

**Comment by Jarcho** [comment]
This shouldn't lint for the same reason we don't lint other external macros. The author of the crate consuming the macro can't necessarily change the code generated by the macro, nor can they easily allow the lint within the macro.

**Comment by winterqt** [comment]
Right, makes sense, thanks.
